src/main/java/com/misterfish/utils/ActionMapper.java

```diff
@@ -0,0 +1,105 @@
+package com.misterfish.utils;
+
+import com.misterfish.config.ModConfigs;
+import com.misterfish.exception.InvalidActionException;
+import com.misterfish.exception.InvalidIntervalException;
+import com.misterfish.exception.InvalidOffsetException;
+import com.misterfish.exception.UnavailableActionException;
+import com.misterfish.helper.EntityPlayerActionPack;
+import it.unimi.dsi.fastutil.Pair;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.network.chat.Component;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+public class ActionMapper {
+    public static EntityPlayerActionPack.ActionType getActionType(String action) {
+        return ACTION_TYPE_MAPPINGS.stream()
+                .filter(entry -> entry.getKey().equalsIgnoreCase(action))
+                .map(Map.Entry::getValue)
+                .findFirst()
+                .orElse(null);
+    }
+
+    public static ArrayList<Pair<EntityPlayerActionPack.ActionType, EntityPlayerActionPack.Action>> getActionPackList(String[] pairs, CommandSourceStack source) {
```

Comment:
Remove source. Mapper should only have string but throw these errors. Try catch where source is available.